src/odemis/driver/smaract.py

```diff
@@ -291,119 +295,203 @@ def __str__(self):
-        self._options = c_char_p("".encode("ascii"))  # In the current version, this must be an empty string.
 
-        for axis_name, axis_par in axes.items():
+        for axis_name, param in axes.items():
+            logging.debug("a %s = %s", axis_name, param)
             try:
-                axis_range = axis_par['range']
+                axis_range = param['range']
             except KeyError:
                 logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
                 axis_range = (-1, 1)
 
             try:
-                axis_unit = axis_par['unit']
+                axis_unit = param['unit']
             except KeyError:
                 logging.info("Axis %s has no unit. Assuming m", axis_name)
                 axis_unit = "m"
 
             ad = model.Axis(canAbs=True, unit=axis_unit, range=axis_range)
             axes_def[axis_name] = ad
-            
-            
+
         # Connect to the device
         self._id = c_uint()
-        self.core.Smarpod_Open(byref(self._id), SmarPodDLL.hwModel, self._locator, self._options)
+        try:
+            self.core.Smarpod_Open(byref(self._id), c_ulong(hwmodel),
+                                   c_char_p(locator.encode("ascii")),
+                                   c_char_p(b""))  # last arg is unused
+        except SmarPodError as ex:
+            if ex.errno == SmarPodDLL.NO_SYSTEMS_FOUND_ERROR:
+                raise model.HwError("Failed to find device, check it is connected and turned on") from ex
+            raise
+
         logging.debug("Successfully connected to SmarPod Controller ID %d", self._id.value)
-        self.core.Smarpod_SetSensorMode(self._id, SmarPodDLL.SMARPOD_SENSORS_ENABLED)
+        self.core.Smarpod_SetSensorMode(self._id, SmarPodDLL.SENSORS_ENABLED)
 
         model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
 
         # Add metadata
-        self._swVersion = self.GetSwVersion()
+        # TODO: GetDLLVersion
+        self._swVersion = "%u.%u.%u" % self.GetDLLVersion()
         self._metadata[model.MD_SW_VERSION] = self._swVersion
         logging.debug("Using SmarPod library version %s", self._swVersion)
 
         self.position = model.VigilantAttribute({}, readonly=True)
+        self._updatePosition()
+
+        # TODO: Change pivot mode to fixed automatically?
+        self.Set_ui(SmarPodDLL.PIVOT_MODE, SmarPodDLL.PIVOT_FIXED)
+        self._metadata[model.MD_PIVOT_POS] = self.GetPivot()
 
         # will take care of executing axis move asynchronously
         self._executor = CancellableThreadPoolExecutor(1)  # one task at a time
 
-        referenced = c_int()
-        self.core.Smarpod_IsReferenced(self._id, byref(referenced))
+        # Use a default actuator speed
+        self.SetSpeed(speed)
+        self.speed = VigilantAttribute({}, unit="m/s", readonly=True)
+        self._updateSpeed()
+        self.SetAcceleration(accel)
+        self._accel = self.GetAcceleration()
```

Comment:
why isn't this a VA?

Reply:
Mostly because we don't have a formal VA in actuators for the acceleration. It wouldn't be hard to think of it, as it'd be the same as the `.speed`. That said here, the accel (like the speed) is so weirdly defined that I don't think we should care too much.